libtelemetry/src/main/java/com/mapbox/android/telemetry/AlarmSchedulerFlusher.java

```diff
@@ -1,35 +1,34 @@
 package com.mapbox.android.telemetry;
 
-
 import android.app.AlarmManager;
 import android.app.PendingIntent;
 import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
+import android.os.Build;
+import android.support.annotation.VisibleForTesting;
 
-import static android.app.PendingIntent.FLAG_CANCEL_CURRENT;
 import static com.mapbox.android.telemetry.SchedulerFlusherFactory.flushingPeriod;
 import static com.mapbox.android.telemetry.SchedulerFlusherFactory.SCHEDULER_FLUSHER_INTENT;
 
 class AlarmSchedulerFlusher implements SchedulerFlusher {
   private final Context context;
   private final AlarmManager manager;
   private final AlarmReceiver receiver;
-  private final int requestCode;
+  private final int REQUEST_CODE = 5328;
   private PendingIntent pendingIntent;
 
-  AlarmSchedulerFlusher(Context context, AlarmManager manager, AlarmReceiver receiver, int requestCode) {
+  AlarmSchedulerFlusher(Context context, AlarmManager manager, AlarmReceiver receiver) {
     this.context = context;
     this.manager = manager;
     this.receiver = receiver;
-    this.requestCode = requestCode;
   }
 
   @Override
   public void register() {
-    Intent alarmIntent = receiver.supplyIntent(requestCode);
-    pendingIntent = PendingIntent.getBroadcast(context, requestCode, alarmIntent, FLAG_CANCEL_CURRENT);
-    String action = SCHEDULER_FLUSHER_INTENT + Integer.toString(requestCode);
+    Intent alarmIntent = receiver.supplyIntent();
+    pendingIntent = PendingIntent.getBroadcast(context, REQUEST_CODE, alarmIntent, 0);
+    String action = SCHEDULER_FLUSHER_INTENT;
```

Comment:
should we just do `IntentFilter filter = new IntentFilter(SCHEDULER_FLUSHER_INTENT);` ?